crates/viewer/re_viewer_context/src/cache/video_stream_cache.rs

```diff
@@ -0,0 +1,587 @@
+        .get_required(&sample_descr)
+        .map_err(|_err| VideoStreamProcessingError::NoVideoSamplesFound)?;
+    let codec_chunks = query_results
+        .get_required(&codec_descr)
+        .map_err(|_err| VideoStreamProcessingError::MissingCodec)?;
+
+    // Translate codec by looking at the last codec.
+    // TODO(andreas): Should validate whether all codecs ever logged are the same, but it's a bit tedious.
+    let last_codec = codec_chunks
+        .last()
+        .and_then(|chunk| chunk.component_instance::<components::VideoCodec>(&codec_descr, 0, 0))
+        .ok_or(VideoStreamProcessingError::MissingCodec)?
+        .map_err(VideoStreamProcessingError::FailedReadingCodec)?;
+    let codec = match last_codec {
+        components::VideoCodec::H264 => re_video::VideoCodec::H264,
+        // components::VideoCodec::H265 => re_video::VideoCodec::H265,
+        // components::VideoCodec::VP8 => re_video::VideoCodec::Vp8,
+        // components::VideoCodec::VP9 => re_video::VideoCodec::Vp9,
+        // components::VideoCodec::AV1 => re_video::VideoCodec::Av1,
+    };
+
+    // Extract all video samples.
+    let mut video_sample_buffers = StableIndexDeque::new();
+    let mut samples = StableIndexDeque::with_capacity(sample_chunks.len()); // Number of video chunks is minimum number of samples.
+    let mut gops = StableIndexDeque::new();
+
+    for chunk in sample_chunks {
+        read_samples_from_chunk(
+            timeline,
+            chunk,
+            codec,
+            &mut video_sample_buffers,
+            &mut samples,
+            &mut gops,
+        )?;
+    }
+
+    let timeline_typ = store
+        .timelines()
+        .get(&timeline)
+        .map_or(TimeType::DurationNs, |t| t.typ());
+    let timescale = match timeline_typ {
+        TimeType::Sequence => re_video::Timescale::IDENTITY, // Can't translate the sequence time to real durations
+        TimeType::DurationNs | TimeType::TimestampNs => re_video::Timescale::NANOSECOND,
+    };
+
+    Ok((
+        re_video::VideoDataDescription {
+            codec,
+            stsd: None,             // Only mp4s have this.
+            coded_dimensions: None, // Unknown so far.
+            timescale,
+            duration: None, // Streams have to be assumed to be open ended, so we don't have a duration.
+            gops,
+            samples,
+            samples_statistics: re_video::SamplesStatistics::NO_BFRAMES, // TODO(#10090): No b-frames for now.
+            mp4_tracks: std::iter::once((0, Some(re_video::TrackKind::Video))).collect(),
+        },
+        video_sample_buffers,
+    ))
+}
+
+/// Reads all video samples from a chunk into an existing video description.
+///
+/// Rejects out of order samples - new samples must have a higher timestamp than the previous ones.
+/// Since samples within a chunk are guaranteed to be ordered, this can only happen if a new chunk
+/// is inserted that has is timestamped to be older than the data in the last added chunk.
```

Comment:
I tink this is wrong. Data within a chunk is sorted by `RowId` which may not be aligned with whatever timeline we are on. In other words: data within a rerun chunk is NOT guaranteed to be sorted. Furthermore, chunks can cover have overlapping time ranges.